Lib/test/support/strace_helper.py

```diff
@@ -0,0 +1,173 @@
+import re
+import sys
+import textwrap
+import unittest
+from dataclasses import dataclass
+from test import support
+from test.support.script_helper import run_python_until_end
+from typing import Dict, List
+
+_strace_binary = "/usr/bin/strace"
+_syscall_regex = re.compile(
+    r"(?P<syscall>[^(]*)\((?P<args>[^)]*)\)\s*[=]\s*(?P<returncode>.+)")
+_returncode_regex = re.compile(
+    r"\+\+\+ exited with (?P<returncode>\d+) \+\+\+")
+
+# Cached value of whether or not there is a compatible strace binary
+_strace_working: bool | None = None
+
+
+@dataclass
+class StraceEvent:
+    syscall: str
+    args: List[str]
+    returncode: str
+
+
+@dataclass
+class StraceResult:
+    strace_returncode: int
+    python_returncode: int
+    _raw_events: str
+    stdout: str
+    stderr: str
+
+    def events(self) -> List[StraceEvent]:
+        """Extract the call list information from _raw_events"""
+        matches = [
+            _syscall_regex.match(event)
+            for event in self._raw_events.splitlines()
+        ]
+        return [
+            StraceEvent(match["syscall"],
+                        [arg.strip() for arg in (match["args"].split(","))],
+                        match["returncode"]) for match in matches if match
+        ]
+
+    def sections(self) -> Dict[str:List[StraceEvent]]:
```

Comment:
This isn't a valid annotation.  Use `-> dict[str, list[StraceEvent]]`...  But it'd better to just not annotate this file at all.  Mention what gets returned in doc strings instead.

BUT This is a good example of why we generally do not have type annotations anywhere within CPython. We are not setup to run multiple type checkers over the stdlib to validate them.  You can see what little is configured in .github/workflows/mypy.yml - but do _not_ try to use this PR to enable it for test.support.  it isn't ready.

[PEP 649](https://peps.python.org/pep-0649/) being implemented is the only reason this was not a Python syntax error.